distribution/zip/jballerina/bin/bal

```diff
@@ -262,6 +262,28 @@ elif [[ $1 == "run" ]] && [[ $2 == *.jar ]]; then # if 1st arg is "run" and 2nd
         -Dballerina.target=jvm \
         -Djava.command=$JAVACMD \
         -jar "${@:2}" # ignores "run"
+
+# running executable(jar) in debug mode
+# if 1st arg is "run", 2nd arg is "--debug", 3rd arg is <debugPort> and 4th arg is executable(jar).
+elif [[ $1 == "run" ]] && [[ $2 == "--debug" ]] && [[ $4 == *.jar ]]; then
+      $JAVACMD \
+        -Xbootclasspath/a:"$BALLERINA_XBOOTCLASSPATH" \
+        -Xms256m -Xmx1024m \
+        -XX:+HeapDumpOnOutOfMemoryError \
+        -XX:HeapDumpPath="$BALLERINA_HOME/heap-dump.hprof" \
+        $JAVA_OPTS \
+        -classpath "$BALLERINA_CLASSPATH" \
+        -Dballerina.home="$BALLERINA_HOME" \
+        -Dballerina.version=2.0.0-beta.1-SNAPSHOT \
+        -Djava.util.logging.config.class="org.ballerinalang.logging.util.LogConfigReader" \
+        -Djava.util.logging.manager="org.ballerinalang.logging.BLogManager" \
+        -Denable.nonblocking=false \
+        -Djava.security.egd=file:/dev/./urandom \
+        -Dfile.encoding=UTF8 \
+        -Dballerina.target=jvm \
+        -Djava.command=$JAVACMD \
+        -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=$3 \
+        -jar "${@:4}" # ignores "run"
```

Comment:
Let's move this logic to the above 'else if' condition, to reduce code duplications

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/30363/commits/6a6cc870877c09e0f9dad8a6149cf726df15019c